controllers/storageclassclaim/storageclassclaim_controller.go

```diff
@@ -92,14 +121,241 @@ func (r *StorageClassClaimReconciler) Reconcile(ctx context.Context, request rec
+			r.log.Info("Finalizer not found for StorageClassClaim. Adding finalizer.", "StorageClassClaim", storageClassClaimRef)
+			r.storageClassClaim.SetFinalizers(append(r.storageClassClaim.GetFinalizers(), StorageClassClaimFinalizer))
+			if err := r.update(r.storageClassClaim); err != nil {
+				return reconcile.Result{}, fmt.Errorf("failed to update StorageClassClaim [%v] with finalizer: %s", storageClassClaimRef, err)
+			}
+		}
+
+		// Creation phase.
+		if r.storageClassClaim.Status.Phase == "" {
+			r.storageClassClaim.Status.Phase = ocsv1alpha1.StorageClassClaimCreating
+
+			// storageClassClaimStorageType is the storage type of the StorageClassClaim
+			var storageClassClaimStorageType providerclient.StorageType
+			switch r.storageClassClaim.Spec.Type {
+			case "blockpool":
+				storageClassClaimStorageType = FulfillstorageclassclaimrequestBlockpool
+			case "sharedfilesystem":
+				storageClassClaimStorageType = FulfillstorageclassclaimrequestSharedfilesystem
+			}
+
+			// Call the `FulfillStorageClassClaim` service on the provider server with StorageClassClaim as a request message.
+			_, err = providerClient.FulfillStorageClassClaim(
+				r.ctx,
+				r.storageCluster.Status.ExternalStorage.ConsumerID,
+				r.storageClassClaim.Name,
+				r.storageClassClaim.Spec.EncryptionMethod,
+				storageClassClaimStorageType,
+			)
+			if err != nil {
+				// Reset StorageClassClaim phase on error since we want the reconciliation loop to enter this block the next time.
+				r.storageClassClaim.Status.Phase = ""
+				return reconcile.Result{}, fmt.Errorf("failed to initiate fulfillment of StorageClassClaim: %v", err)
+			}
+		}
+		r.storageClassClaim.Status.Phase = ocsv1alpha1.StorageClassClaimConfiguring
+
+		// Configuration phase
+		// Triggered only if the corresponding StorageClass is not found.
+		// TODO Possible to check for Secret, in addition to existing StorageClass check (secret name is dynamic)?
+		correspondingStorageClassDoesNotExists := errors.IsNotFound(
+			r.get(&storagev1.StorageClass{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: r.storageClassClaim.Name,
+				},
+			}),
+		)
+		if r.storageClassClaim.Status.Phase == ocsv1alpha1.StorageClassClaimConfiguring && correspondingStorageClassDoesNotExists {
+
+			// Call the `GetStorageClassClaimConfig` service on the provider server with StorageClassClaim as a request message.
+			response, err := providerClient.GetStorageClassClaimConfig(
+				r.ctx,
+				r.storageCluster.Status.ExternalStorage.ConsumerID,
+				r.storageClassClaim.Name,
+			)
+			if err != nil {
+				return reconcile.Result{}, fmt.Errorf("failed to get StorageClassClaim config: %v", err)
+			}
+			resources := response.ExternalResource
+			if resources == nil {
+				return reconcile.Result{}, fmt.Errorf("no configuration data recieved")
+			}
+
+			// Go over the received objects and operate on them accordingly.
+			for _, resource := range resources {
+				data := map[string][]byte{}
```

Comment:
Why is this a `map[string]byte{}` and not `map[string]string`? 
The current ocs-operator code for Provider/consumer mode is using `map[string]string` (https://github.com/red-hat-storage/ocs-operator/blob/4f51ba4a537a9209208ef141b85b6438080ba8d9/controllers/storagecluster/external_ocs.go#L179-L180) and a lot of the code below needs this to be `map[string]string` in order to work.